generators/entity-client/templates/react/src/main/webapp/app/entities/_entity.tsx

```diff
@@ -45,11 +56,25 @@ export interface I<%= entityReactName %>Props {
   <%_ } _%>
   match: any;
 }
+<%_ if (searchEngine === 'elasticsearch') { _%>
+
+export interface I<%= entityReactName %>State {
+  search: string;
+}
+<%_ } _%>
 
-export class <%= entityReactName %> extends React.Component<I<%= entityReactName %>Props, undefined> {
+export class <%= entityReactName %> extends React.Component<I<%= entityReactName %>Props, I<%= entityReactName %>State> {
 
   constructor(props) {
     super(props);
+    <%_ if (searchEngine === 'elasticsearch') { _%>
+    this.state = {
+      search: ''
+    };
+    this.search = this.search.bind(this);
```

Comment:
no need to bind just use arrow methods instead

Reply:
there is a lint warning when I use arrow methods, so I use this ...